### PR TITLE
Corrected limiting the data vector size in test_reduce and test_trans…

### PR DESCRIPTION
**Pull request by kpyzhov**
…form.

### CODE REVIEW

**Review by stanleytsang-amd — APPROVED** [review]
We should wait for @neon60's feedback before merging since he wrote the test. An explanation for Istvan's benefit:
The root cause of the problem is if "T" is a floating point type, the value of std::numeric_limits<T>::max() converted to size_t is undefined value. 

Compiler team was seeing test failures with transform and reduce with newer builds because size/n was getting set to bad values, I was also seeing failures with HIP on Windows.

**Review by neon60 — APPROVED** [review]

**Comment by neon60** [comment]
> We should wait for @neon60's feedback before merging since he wrote the test. An explanation for Istvan's benefit: The root cause of the problem is if "T" is a floating point type, the value of std::numeric_limits::max() converted to size_t is undefined value.
> 
> Compiler team was seeing test failures with transform and reduce with newer builds because size/n was getting set to bad values, I was also seeing failures with HIP on Windows.

I will write out an issue to handle half and bfloat16 differently. Checked with different compilers the ```static_cast<size_t>(std::numeric_limits<float>::max())``` and I saw lot of different results. The change looks good.

**Comment by kpyzhov** [comment]
Thanks！

**Pull request merged**
